File: 0002-两数相加/Solution.java
class ListNode {
    int val;
    ListNode next;

    ListNode(int x) {
        val = x;
    }
}

class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {   
        int fbi = 0;
        ListNode head = new ListNode(0);
        ListNode work = head;
        while(l1 != null || l2 != null || fbi == 1){
            if(l1 != null){
                fbi += l1.val;
                l1 = l1.next;
            }
            if(l2 != null){
                fbi += l2.val;
                l2 = l2.next;
            }
            work.next = new ListNode(fbi % 10);
            work = work.next;
            fbi = (fbi >= 10? 1: 0);
        }
        return head.next;
    }
}

File: 0002-两数相加/Solution.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution:
    def addTwoNumbers(self, l1: ListNode, l2: ListNode) -> ListNode:
        ret = ListNode(None, None)
        p, flag = ret, 0
        while l1 or l2:
            value = flag
            if l1:
                value += l1.val
                l1 = l1.next
            if l2:
                value += l2.val
                l2 = l2.next
            value, flag = value % 10, value // 10
            p.next = ListNode(value)
            p = p.next
        if flag != 0:
            p.next = ListNode(flag, None)
        return ret.next
